.eslintrc

```diff
@@ -41,77 +29,120 @@
+                "message": "Please use Object.defineProperty instead."
+            },
+            {
+                "property": "__defineSetter__",
+                "message": "Please use Object.defineProperty instead."
+            },
+            {
+                "object": "Math",
+                "property": "pow",
+                "message": "Use the exponentiation operator (**) instead."
+            },
+            {
+                "object": "globalThis",
+                "property": "lwcRuntimeFlags",
+                "message": "Use the bare global lwcRuntimeFlags instead."
+            }
+        ],
         "no-self-compare": "error",
         "no-undef-init": "error",
         "no-useless-computed-key": "error",
         "no-useless-return": "error",
-        "prefer-const": ["error", {
-            "destructuring": "any",
-            "ignoreReadBeforeAssign": true
-        }],
+        "prefer-const": [
+            "error",
+            {
+                "destructuring": "any",
+                "ignoreReadBeforeAssign": true
+            }
+        ],
         "template-curly-spacing": "error",
         "yoda": "error",
 
         "@lwc/lwc-internal/no-invalid-todo": "error",
         "import/order": [
             "error",
-            { "groups": ["builtin", "external", "internal", "parent", "index", "sibling", "object", "type"] }
+            {
+                "groups": [
+                    "builtin",
+                    "external",
+                    "internal",
+                    "parent",
+                    "index",
+                    "sibling",
+                    "object",
+                    "type"
+                ]
+            }
         ],
-        "no-restricted-imports": ["error", {
-            "name": "@lwc/features",
-            "importNames": ["lwcRuntimeFlags", "runtimeFlags", "default"],
-            "message": "Do not directly import runtime flags from @lwc/features. Use the global lwcRuntimeFlags variable instead."
-        }]
+        "no-restricted-imports": [
+            "error",
+            {
+                "name": "@lwc/features",
+                "importNames": ["lwcRuntimeFlags", "runtimeFlags", "default"],
+                "message": "Do not directly import runtime flags from @lwc/features. Use the global lwcRuntimeFlags variable instead."
+            }
+        ],
+        "header/header": [
+            2,
+            "block",
+            [
+                "",
+                {
+                    "pattern": "^ \\* Copyright \\(c\\) \\d{4}, ([sS]alesforce.com, inc|Salesforce, Inc)\\.$",
+                    "template": " * Copyright (c) 2024, Salesforce, Inc."
+                },
+                " * All rights reserved.",
+                " * SPDX-License-Identifier: MIT",
+                " * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT",
+                " "
+            ]
+        ]
```

Comment:
Here's the new rule. Dunno why VS Code decided that the whole file should be re-formatted. 🤷

Reply:
I'm confused. If we have an ESLint plugin now, then why do we even need the script to check/update the headers? This should be folded into the normal `eslint`/`eslint --fix` process.

I also would expect this PR to contain an update to every file to have the proper (2024) header.

Reply:
To avoid editor reformatting issues, we can use something like [editorconfig](https://editorconfig.org/). I'm surprised we haven't already set this up.

Reply:
OK this was answered verbally – we need to check files that aren't tracked by ESLint, such as .html/.css. Although I think the value of this is debatable since we only really ship .js/.d.ts files.

Reply:
I checked to see how many files contain the word "copyright":
```
  'LICENSE' => 1,
  '.js' => 170,
  '.ts' => 312,
  '.mjs' => 5,
  '.html' => 10,
  '.sh' => 2
```

Given we have 1754 HTML files, I don't think we actually need to enforce this script on 10 of them. I'll remove it.

Reply:
Perfect. js/ts/mjs should cover everything important here.

Reply:
Turns out all the formatting changes are because nobody's touched this file with prettier in a while, and I guess prettier's updated their rules since then. I bypassed this for now, though, so that the diff is just my actual changes.